lib/src/third_party/diff_match_patch/diff.dart

```diff
@@ -0,0 +1,990 @@
+// Diff Match and Patch
+// Copyright 2018 The diff-match-patch Authors.
+// https://github.com/google/diff-match-patch
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:collection';
+import 'dart:math';
+
+import 'package:meta/meta.dart';
+
+part 'utils.dart';
+
+enum Operation { delete, insert, equal }
+
+/// Class representing one diff operation.
+class Diff {
+  /// One of: Operation.insert, Operation.delete or Operation.equal.
+  Operation operation;
+
+  /// The text associated with this diff operation.
+  String text;
+
+  /// Constructor.  Initializes the diff with the provided values.
+  ///
+  /// [operation] is one of Operation.insert, Operation.delete or Operation.equal.
+  /// [text] is the text being applied.
+  Diff(this.operation, this.text);
+
+  /// Display a human-readable version of this Diff.
+  /// Returns a text version.
+  @override
+  String toString() {
+    var prettyText = text.replaceAll('\n', '\u00b6');
+    return 'Diff($operation,"$prettyText")';
+  }
+
+  /// Is this Diff equivalent to another Diff?
+  /// [other] is another Diff to compare against.
+  /// Returns true or false.
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is Diff &&
+          runtimeType == other.runtimeType &&
+          operation == other.operation &&
+          text == other.text;
+
+  /// Generate a uniquely identifiable hashcode for this Diff.
+  /// Returns numeric hashcode.
+  @override
+  int get hashCode => operation.hashCode ^ text.hashCode;
+}
+
+class DiffMatchPatch {
+  double diffTimeout = 1.0;
```

Comment:
Why about we just make this a global variable... and then we don't need the `DiffMatchPatch` wrapper class?

Reply:
Do we even need `diffTimeout`?